File: app/src/test/java/com/hamboomger/model/event/EventAgendaTest.java
package com.hamboomger.model.event;

import org.junit.Test;

import java.util.Arrays;

import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.assertThat;

/**
 * @author ddorochov
 */
public class EventAgendaTest {
	@Test
	public void testAssembling() {
		// arrange
		String midnightAppointment = "Midnight event";
		String noonAppointment = "Noon event";

		// act
		EventAgenda agenda = new EventAgenda(Arrays.asList(midnightAppointment, noonAppointment));

		// assert
		assertThat(agenda.getAppointments().size(),
				is(2));
	}
}

File: app/src/test/java/com/hamboomger/model/event/EventAgendaTest.kt
package com.hamboomger.model.event

import org.junit.Assert.*
import org.junit.Test
import java.time.LocalTime

/**
 * @author ddorochov
 */
class EventAgendaTest {
    @Test fun testAssembling() {
        // arrange
        val midnightAppointment = Appointment(LocalTime.MIDNIGHT, "Midnight event")
        val noonAppointment = Appointment(LocalTime.NOON, "Noon event")

        // act
        val builder = EventAgenda.Builder()
                .addAppointment(midnightAppointment)
                .addAppointment(noonAppointment)
                .addAdditionalInfo("A")
                .addAdditionalInfo("B")
                .addAdditionalInfo("C")
        val agenda = builder.build()

        // assert
        assertTrue(agenda.additionalInfo == "A\nB\nC\n")
        assertTrue(agenda.appointments.size == 2)
    }
}